src/SFML/Window/Win32/JoystickImpl.cpp

```diff
@@ -45,1114 +45,717 @@
-            result = m_device->SetProperty(DIPROP_BUFFERSIZE, &property.diph);
+    for (DWORD xinputIndex = 0; xinputIndex < XINPUT_MAX_CONTROLLERS; xinputIndex++)
+    {
+        XINPUT_STATE xinputState  = {0};
+        auto xinputResult    = XInputGetState(xinputIndex, &xinputState);
+        if (xinputResult != 0x0)
+        {
+            // Probably not connected, just move on.
+            continue;
+        }
 
-            if (result == DI_OK)
-            {
-                // Buffering supported
-                m_buffered = true;
-            }
-            else if (result == DI_POLLEDDEVICE)
-            {
-                // Only polling supported
-                m_buffered = false;
-            }
-            else
+        for (DWORD joystickIndex = 0; joystickIndex < m_sJoysticks.size(); joystickIndex++)
+        {
+            JoystickImpl& joystick = m_sJoysticks.at(joystickIndex);
+            if (joystick.m_xInputIndex == xinputIndex)
             {
-                err() << "Failed to set DirectInput device buffer size for device "
-                      << std::quoted(m_identification.name.toAnsiString()) << ": " << result << std::endl;
-
-                m_device->Release();
-                m_device = nullptr;
+                if (joystick.m_xInputPacketNumber != xinputState.dwPacketNumber)
+                {
+                    joystick.m_xInputPacketNumber = xinputState.dwPacketNumber;
+
+                    auto& state       = joystick.m_state;
+                    auto& gamepad     = xinputState.Gamepad;
+                    state.buttons[0]  = !!(gamepad.wButtons & XINPUT_GAMEPAD_A);
+                    state.buttons[1]  = !!(gamepad.wButtons & XINPUT_GAMEPAD_B);
+                    state.buttons[2]  = !!(gamepad.wButtons & XINPUT_GAMEPAD_X);
+                    state.buttons[3]  = !!(gamepad.wButtons & XINPUT_GAMEPAD_Y);
+                    state.buttons[4]  = !!(gamepad.wButtons & XINPUT_GAMEPAD_DPAD_UP);
+                    state.buttons[5]  = !!(gamepad.wButtons & XINPUT_GAMEPAD_DPAD_DOWN);
+                    state.buttons[6]  = !!(gamepad.wButtons & XINPUT_GAMEPAD_DPAD_LEFT);
+                    state.buttons[7]  = !!(gamepad.wButtons & XINPUT_GAMEPAD_DPAD_RIGHT);
+                    state.buttons[8]  = !!(gamepad.wButtons & XINPUT_GAMEPAD_START);
+                    state.buttons[9]  = !!(gamepad.wButtons & XINPUT_GAMEPAD_BACK);
+                    state.buttons[10] = !!(gamepad.wButtons & XINPUT_GAMEPAD_LEFT_SHOULDER);
+                    state.buttons[11] = !!(gamepad.wButtons & XINPUT_GAMEPAD_RIGHT_SHOULDER);
+                    state.buttons[12] = !!(gamepad.wButtons & XINPUT_GAMEPAD_LEFT_THUMB);
+                    state.buttons[13] = !!(gamepad.wButtons & XINPUT_GAMEPAD_RIGHT_THUMB);
+
+                    SHORT deadzone = XINPUT_GAMEPAD_LEFT_THUMB_DEADZONE / 4;
+
+                    state.axes[Joystick::Axis::X] = (abs(gamepad.sThumbLX) < deadzone) ? 0.0f : gamepad.sThumbLX / 327.670f;
```

Comment:
Magic number of `327.670` here, can we convert this to a const with some context as to what this value means in the context of the conversion being done here